pkg/apis/componentconfig/validation/validation_pluginargs.go

```diff
@@ -102,3 +109,80 @@ func ValidateRemoveDuplicatesArgs(args *componentconfig.RemoveDuplicatesArgs) er
 		validateNamespaceArgs(args.Namespaces),
 	)
 }
+
+// validateThresholds checks if thresholds have valid resource name and resource percentage configured
+func validateThresholds(thresholds api.ResourceThresholds) error {
+	if len(thresholds) == 0 {
+		return fmt.Errorf("no resource threshold is configured")
+	}
+	for name, percent := range thresholds {
+		if percent < MinResourcePercentage || percent > MaxResourcePercentage {
+			return fmt.Errorf("%v threshold not in [%v, %v] range", name, MinResourcePercentage, MaxResourcePercentage)
+		}
+	}
+	return nil
+}
+
+func ValidateHighUtilizationPluginConfig(thresholds, targetThresholds api.ResourceThresholds) error {
+	if targetThresholds != nil {
+		return fmt.Errorf("targetThresholds is not applicable for HighNodeUtilization")
+	}
+	if err := validateThresholds(thresholds); err != nil {
+		return fmt.Errorf("thresholds config is not valid: %v", err)
+	}
+	return nil
+}
+
+func ValidateHighNodeUtilizationArgs(args *componentconfig.HighNodeUtilizationArgs) error {
+	var err error
+
+	return errorsAggregate(
+		err,
+		ValidateHighUtilizationPluginConfig(args.NodeResourceUtilizationThresholds.Thresholds, args.NodeResourceUtilizationThresholds.TargetThresholds),
+		validateThresholds(args.NodeResourceUtilizationThresholds.Thresholds),
+	)
+}
+
+func ValidateLowNodeUtilizationArgs(args *componentconfig.LowNodeUtilizationArgs) error {
+	var err error
+
+	return errorsAggregate(
+		err,
+		validateNodeUtilization(args),
+		ValidateLowUtilizationPluginConfig(args.NodeResourceUtilizationThresholds.Thresholds, args.NodeResourceUtilizationThresholds.TargetThresholds, args.NodeResourceUtilizationThresholds.UseDeviationThresholds),
+	)
+
+}
+func validateNodeUtilization(args *componentconfig.LowNodeUtilizationArgs) error {
+	if args == nil || args.NodeResourceUtilizationThresholds == nil {
```

Comment:
technically args cannot be nil

Reply:
good idea, i update as your mention, please check